pkg/virt-api/rest/subresource.go

```diff
@@ -628,13 +628,13 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 		}
 		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("%v does not support manual stop requests", v1.RunStrategyHalted)), response)
 		return
-	case v1.RunStrategyManual:
+	case v1.RunStrategyManual, v1.RunStrategyRerunOnFailure:
 		if !hasVMI || vmi.IsFinal() {
 			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf(vmNotRunning)), response)
 			return
 		}
 		// pass the buck and ask virt-controller to stop the VM. this way the
-		// VM will retain RunStrategy = manual
+		// VM will retain RunStrategy = manual or RerunOnFalilure
```

Comment:
Got it
@davidvossel so do you have any advice for this purpose?
I found i should not stop and start with paused that is a little bit different from what we want

Reply:
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L645

according to this line
virt-controller will not start VMI again if status request has stopRequest?

Reply:
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L640-L677

there are some conditions that we don't consider?
it seems forceStop didn't work properly here.
because if we forceStop RerunOnFailure VM.
it will delete VMI and next time to execute startStop, it will consider as VMI is gone and fail, so restart the VMI again.